### PR TITLE
Deploy both the OCR-API and frontend together with reusable workflows

### PR DESCRIPTION
**Pull request by derekadombek**
## Description
This entire build and deploy workflow is coming together. The last step is to add deployment to it.

## Related Issues
[Link any related issues or tasks from your project management system.]

## Checklist
- [x] The title of this PR is descriptive and concise.
- [x] My changes follow the style guidelines of this project.
- [x] I have added or updated test cases to cover my changes.
- [ ] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: .github/actions/build-publish-api/action.yml [hunk 1]
```diff
@@ -26,8 +26,6 @@ inputs:
 runs:
   using: composite
   steps:
-      - name: Checkout repository
-        uses: actions/checkout@v4
```
Comment (derekadombek) [inline .github/actions/build-publish-api/action.yml:30]: don't need this cause we are already checking out the repo code from the base `deploy-dev.yml` file

File: .github/actions/build-frontend/action.yml [hunk 2]
```diff
@@ -1,9 +1,6 @@
 name: Build ReportVision's Front End
 description: Build the React application
 inputs:
-  deploy-env:
-    description: The environment being deployed (e.g. "prod" or "test")
-    required: true
```
Comment (derekadombek) [inline .github/actions/build-frontend/action.yml:6]: this variable was never being used.

File: .github/actions/build-frontend/action.yml [hunk 3]
```diff
@@ -16,13 +13,16 @@ inputs:
   api-endpoint:
     description: The endpoint to connect the frontend to an api
     required: true
+  node-version:
+    description: Node.js Version
```
Comment (derekadombek) [inline .github/actions/build-frontend/action.yml:17]: creating this new var so we dont have to set a global var and so we can make this action more reusable for other situations later on possibly.

File: .github/workflows/deploy-dev.yml [hunk 4]
```diff
@@ -76,4 +71,28 @@ jobs:
...
           azure-tenant-id: ${{ secrets.AZURE_TENANT_ID }}
           azure-subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-          app-name: reportvision
\ No newline at end of file
+          app-name: reportvision
+
+  deploy-services:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.deploy-env }}
+    needs: [build-publish-ocr, build-frontend, environment-setup]
```
Comment (derekadombek) [inline .github/workflows/deploy-dev.yml:79]: deploying both the frontend and ocr-api only after they are both built and the infrastructure is setup with Terraform.

File: .github/actions/build-frontend/action.yml (hunk 3 above)
Comment (schreiaj) [reply .github/actions/build-frontend/action.yml:17]: Is this really something we want in an action vs as a config in the docker file? I can't see a place where we'd want to just bump node version

**Review by schreiaj — APPROVED** [review]
I'm not sold on the Node Version thing, but if there's a use case that makes sense I'm happy to defer to your expertise there. Otherwise LGTM

File: .github/actions/build-frontend/action.yml (hunk 3 above)
Comment (derekadombek) [reply .github/actions/build-frontend/action.yml:17]: I agree! we should just do that in docker for sure. However currently we are currently not containerizing our frontend in the live environments, we're using just blob storage which requires us to npm build in CICD and push the build files to the Blob. I think the direction you Arin and I would like to go will change this is the future for prod environments and we can finally dockerize it.

**Pull request merged**